server/reload.go

```diff
@@ -115,41 +233,86 @@ func (s *Server) diffOptions(newOpts *Options) ([]option, error) {
 		}
 		switch strings.ToLower(field.Name) {
```

Comment:
This will get huge...   maybe break apart into functions - e.g. `handleTraceOptions(...)`, `handleTLSAuthOptions(..)`, `handleUserAuthOptions(...)`